PlutoHelperAgent/AppDelegate.m

```diff
@@ -92,6 +110,9 @@ - (void)applicationDidFinishLaunching:(NSNotification *)aNotification {
 
 - (void)applicationWillTerminate:(NSNotification *)aNotification {
     // Insert code here to tear down your application
+    NSLog(@"Application about to quit.");
+    [SharedFunctions logout_of_project_server];
+    sleep(1);
```

Comment:
can probably drop this `sleep()` for production code

Reply:
The sleep is required so that function above completes. If it did not exist the programme may quit before all the code it run.

Reply:
@Dala0 hmm, what happens if the above line takes longer to complete than 1 second?  If `logout_of_project_server` is doing an async operation then surely some token is returned that you can wait for?

Reply:
If it takes longer than a second then the function may not complete. One second was more than enough in my tests, so I have left it like this.